### PR TITLE
BOAC-112 Fix KeyError in analytics lib; better error handling generally

### PR DESCRIPTION
**Pull request by paulkerschen**
https://jira.ets.berkeley.edu/jira/browse/BOAC-112

Fix the observed error. Also log error API responses at a higher level, and handle those responses on the cohort front end (`w=1` is recommended to review this last).

### CODE REVIEW

File: boac/static/app/cohort/cohort.html [hunk 1]
```diff
@@ -4,27 +4,32 @@
   <h1 data-ng-bind="cohort.name"></h1>
 
   <h2>Student Performance Assessment Graph</h2>
-  <div class="container" data-ng-if="!cohort">
-    <div>Loading.... </div>
+  <div class="container" data-ng-if="errored">
+    There was an error retriving cohort data.
```
Comment (johncrossman) [inline boac/static/app/cohort/cohort.html]: @paulkerschen I belive "retriving" is misspelled.

File: boac/static/app/cohort/cohort.html [hunk 2]
```diff
@@ -4,27 +4,32 @@
...
-    <h4>Students without graphable data</h4>
-    <li data-ng-repeat="member in membersWithoutData">
-      <a data-ng-href="/student/{{member.uid}}">
-        <img class="cohort-avatar" data-ng-src="{{member.avatar_url}}">
-        <span data-ng-bind="member.name"></span>
-      </a>
-    </li>
-  </ul>
+  <div class="container" data-ng-if="!errored && cohort">
+    <ul data-ng-if="membersWithoutData.length">
```
Comment (johncrossman) [inline boac/static/app/cohort/cohort.html]: @paulkerschen  any chance we rename var to 'menWithoutHats'?

**Comment by johncrossman** [comment]
Looks good, @paulkerschen  Only a typo to fix.

**Comment by paulkerschen** [comment]
Thanks @johncrossman; retrieve reviesed.

**Pull request merged**